diff --git a/field_collection.module b/field_collection.module
index 9336f7e..3c4d278 100644
--- a/field_collection.module
+++ b/field_collection.module
@@ -1532,7 +1532,8 @@ function field_collection_field_widget_embed_validate($element, &$form_state, $c
 
       // Copied from _form_validate().
       if (isset($elements['#needs_validation'])) {
-        $is_empty_multiple = (!count($elements['#value']));
+        $is_countable = is_array($elements['#value']) || $elements['#value'] instanceof Countable;
+        $is_empty_multiple = $is_countable && (!count($elements['#value']));
         $is_empty_string = (is_string($elements['#value']) && drupal_strlen(trim($elements['#value'])) == 0);
         $is_empty_value = ($elements['#value'] === 0);
         $is_empty_option = (isset($elements['#options']['_none']) && $elements['#value'] == '_none');
