diff --git a/handlers/views_handler_filter.inc b/handlers/views_handler_filter.inc
index 8583605..edeb72b 100644
--- a/handlers/views_handler_filter.inc
+++ b/handlers/views_handler_filter.inc
@@ -744,7 +744,7 @@ class views_handler_filter extends views_handler {
           $form[$value]['#multiple'] = TRUE;
         }
         unset($form[$value]['#default_value']);
-        if (empty($form_state['input'])) {
+        if (empty($form_state['input'][$value])) {
           $form_state['input'][$value] = $this->group_info;
         }
       }
